contracts/IBilling.sol

```diff
@@ -36,12 +36,17 @@ interface IBilling {
     function addTo(address _to, uint256 _amount) external;
 
     /**
-     * @dev Add tokens into the billing contract for any user, from L1
-     * This can only be called from L2GraphTokenGateway.finalizeInboundTransfer.
-     * @param _user  Address that tokens are being added to
-     * @param _amount  Amount of tokens to add
+     * @dev Receive tokens with a callhook from the Arbitrum GRT bridge
+     * Expects an `address user` in the encoded _data.
+     * @param _from Token sender in L1
+     * @param _amount Amount of tokens that were transferred
+     * @param _data ABI-encoded callhook data: contains address that tokens are being added to
      */
-    function addFromL1(address _user, uint256 _amount) external;
+    function onTokenTransfer(
```

Comment:
In our main contracts repo, we could include a reference `ICrossChainTokenReceiver` that contracts can implement.

Reply:
Good call, I did add an `ICallhookReceiver`: https://github.com/graphprotocol/contracts/pull/719/files#diff-fe53aefef99df313ce11d09b1afd063084819f62b3cfb79e061b4baa062fa7f6 - I just didn't use it here because I didn't want to add a `@graphprotocol/contracts` dependency just for that...